services/DataApiQueueService.cfc

```diff
@@ -84,7 +84,7 @@ component {
 
 								dataEntry.record = _aliasFields( record.object_name, recordData );
 							} catch( any e ) {
-								dataEntry.record = record.data;
+								dataEntry.record = isStruct( recordData ?: "" ) ? _aliasFields( record.object_name, recordData ) : record.data;
```

Comment:
I'm not sure this is right - feels a bit of a funge.

I think I would expect that the try/catch is purely for the SerializeJson(), but could probably be done cleaner with IsJson()? So:

```cfc
var recordData = IsJson( record.data ) ? DeserializeJson( record.data ) : record.data;

if ( IsStruct( recordData ) ) {
	if ( $isFeatureEnabled( "dataApiFormulaFieldsForAtomic" ) ) {
		var formulaFields = configSvc.getEntityFormulaFields( entity=entity );

		if ( ArrayLen( formulaFields ) ) {
			StructAppend( recordData, apiSvc.getSingleRecord( entity=entity, recordId=record.record_id, fields=formulaFields ), false );
		}
	}

	dataEntry.record = _aliasFields( record.object_name, recordData );
}
```

Does this seem right to you?